core/state_processor.go

```diff
@@ -71,6 +71,9 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	if p.config.DAOForkSupport && p.config.DAOForkBlock != nil && p.config.DAOForkBlock.Cmp(block.Number()) == 0 {
 		misc.ApplyDAOHardFork(statedb)
 	}
+	if header.BeaconRoot != nil {
+		misc.ApplyBeaconRoot(header, statedb)
+	}
```

Comment:
Wouldn't this be a pretty good place to verify that the existence of `BeaconRoot` matches our expectation, depending on what fork rules we're operating under?